config/routes.rb

```diff
@@ -1,6 +1,37 @@
 Rails.application.routes.draw do
+  devise_for :users
+
+  get '/users', to: 'users#index', as: 'splash'
+  get '/users', to: 'users#index', as: 'user'
+
+  
+  root 'categories#index', as: "categorytrack"
+  
+  resources :users, only: [:index]
+  resources :categories
+  resources :expenses
+  
+  devise_scope :user do
+    authenticated :user do
+      root "categories#index", as: "authenticated_root"
+      get '/users/sign_out' => 'devise/sessions#destroy'
+    end
+    unauthenticated :user do
+      root "users#index"
+    end
+  end
+  
+  # unauthenticated do
+  #   root "users#index"
+  # end
+
+  # authenticated do
+  #   root "categories#index", as: "authenticated_root"
+  #   get '/users/sign_out' => 'devise/sessions#destroy'
+  # end
+  
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
-  # root "articles#index"
+  # root "categories#index", as: "category_track"
```

Comment:
- [ ] To best practices Do not commit old pieces of code as inline comments. Please delete it. [Requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md#:~:text=Do%20not%20commit%20old%20pieces%20of%20code%20as%20inline%20comments.%20They%20will%20make%20your%20project%20look%20messy.%20If%20you%20need%20to%20review%20a%20previous%20version%20of%20your%20code%2C%20you%20can%20always%20use%20git%20history.)